ui/src/components/chat/compose.rs

```diff
@@ -465,20 +464,28 @@ fn get_messages(cx: Scope<ComposeProps>) -> Element {
                     let active_chat = data.active_chat.clone();
                     let last_message = messages.last().unwrap().message.clone();
                     let sender = state.read().get_identity(&group.sender);
+                    let sender_name = sender.username();
                     let active_language = state.read().settings.language.clone();
                     let platform = sender.platform().into();
-                    let status = convert_status(&sender.identity_status());
+
+                    // todo: investigate why updating one's profile picture affects their status. 
+                    // this is probably in warp but I'm not sure. 
+                    let mut sender_status = sender.identity_status().into();
+                    if !group.remote && sender_status == Status::Offline {
```

Comment:
Besides the use of metadata, the status itself is determine by the actual connectivity status so updating, it shouldnt change it unless for whatever reason the `Identity` wrapper status is defaulting back to offline here in uplink.